sdk/tests/resources/tex-input-validation.js

```diff
@@ -394,4 +429,98 @@ for (var ii = 0; ii < testCases.length; ++ii) {
   testCopyFromInternalFBO(testCases[ii]);
 }
 
+if (contextVersion > 1) {
+// Create new texture for testing api of WebGL 2.0.
+shouldBeNonNull("tex = gl.createTexture()");
+gl.bindTexture(gl.TEXTURE_3D, tex);
+wtu.glErrorShouldBe(gl, gl.NO_ERROR);
+
+debug("");
+debug("Checking TexImage3D: a set of inputs that are valid in GL but invalid in WebGL");
+
+var testCases = [
+  { target: 0x8064, // GL_PROXY_TEXTURE_2D
+    internalFormat: gl.RGBA,
+    border: 0,
+    format: gl.RGBA,
+    type: gl.UNSIGNED_BYTE,
+    expectedError: gl.INVALID_ENUM },
+  { target: gl.TEXTURE_2D,
+    internalFormat: gl.RGBA,
+    border: 0,
+    format: gl.RGBA,
+    type: gl.UNSIGNED_BYTE,
+    expectedError: gl.INVALID_ENUM },
+  { target: gl.TEXTURE_3D,
+    internalFormat: gl.RED,
+    border: 0,
+    format: gl.RED,
+    type: gl.UNSIGNED_BYTE,
+    expectedError: gl.INVALID_ENUM },
```

Comment:
This looks like it fails because the internal format should be gl.RED8, and it looks to me like it should generate INVALID_OPERATION, not INVALID_ENUM. It also looks like it should fail in OpenGL ES as well as WebGL.